auth/grants.go

```diff
@@ -19,16 +19,44 @@ import (
 	"strings"
 
 	"golang.org/x/exp/slices"
+	"google.golang.org/protobuf/encoding/protojson"
 
 	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/utils"
 )
 
+type RoomConfiguration livekit.RoomConfiguration
+
+var tokenMarshaler = protojson.MarshalOptions{
+	EmitDefaultValues: false,
+}
+
+func (c *RoomConfiguration) Clone() *RoomConfiguration {
+	if c == nil {
+		return nil
+	}
+	return (*RoomConfiguration)(utils.CloneProto((*livekit.RoomConfiguration)(c)))
+}
+
+func (c *RoomConfiguration) MarshalJSON() ([]byte, error) {
+	return tokenMarshaler.Marshal((*livekit.RoomConfiguration)(c))
+}
+
+func (c *RoomConfiguration) UnmarshalJSON(data []byte) error {
+	return protojson.Unmarshal(data, (*livekit.RoomConfiguration)(c))
+}
+
 type ClaimGrants struct {
 	Identity string      `json:"-"`
 	Name     string      `json:"name,omitempty"`
 	Kind     string      `json:"kind,omitempty"`
 	Video    *VideoGrant `json:"video,omitempty"`
 	SIP      *SIPGrant   `json:"sip,omitempty"`
+	// Room configuration to use if this participant initiates the room
+	Room *RoomConfiguration `json:"room,omitempty"`
+	// Cloud-only, config preset to use
+	// when both room and roomPreset are set, parameters in room overrides the preset
+	RoomPreset string `json:"roomPreset,omitempty"`
```

Comment:
are we using `roomPreset` or `room_preset` in config?

Reply:
protojson serialization serializes to camelCase by default